arroyo-rpc/src/schema_resolver.rs

```diff
@@ -0,0 +1,167 @@
+use anyhow::{anyhow, bail};
+use apache_avro::Schema;
+use async_trait::async_trait;
+use reqwest::{Client, StatusCode, Url};
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use tracing::warn;
+
+#[async_trait]
+pub trait SchemaResolver: Send {
+    async fn resolve_schema(&self, id: u32) -> Result<Option<String>, String>;
```

Comment:
Why not use anyhow errors?

Reply:
Because the errors are meant to be used as the message in a UserError, which takes a string. You could rely on `anyhow::Error::to_string`, but that includes various formatting logic that I don't want; the SchemaResolver should be responsible for producing a good String error message that can be shown to the user.